ansible_base/rbac/claims.py

```diff
@@ -0,0 +1,306 @@
+            object_arrays[model_type].append(object_data)
+
+        # Get the array position from the cache
+        array_position = ansible_id_to_index[model_type][ansible_id]
+
+        # If the role is not in object_roles, initialize it
+        if role_name not in object_roles:
+            object_roles[role_name] = {'content_type': model_type, 'objects': []}
+
+        # Add the array position to the role
+        object_roles[role_name]['objects'].append(array_position)
+
+    # Resolve team organization references
+    _resolve_team_organization_references(object_arrays, ansible_id_to_index['organization'])
+
+    return object_arrays, object_roles
+
+
+def _get_user_global_roles(user: Model) -> list[str]:
+    """Get user's global role assignments.
+
+    Args:
+        user: User model instance
+
+    Returns:
+        List of global role names assigned to the user
+
+    Example:
+        ['Platform Auditor', 'System Administrator']
+    """
+    global_roles_query = RoleDefinition.objects.filter(content_type=None, user_assignments__user=user.pk, name__in=settings.ANSIBLE_BASE_JWT_MANAGED_ROLES)
+
+    return [role_definition.name for role_definition in global_roles_query]
+
+
+def get_user_claims(user: Model) -> dict[str, Union[list[str], dict[str, Union[str, list[dict[str, str]]]]]]:
+    """Generate comprehensive claims data for a user including roles and object access.
+
+    This function builds a complete picture of a user's permissions by gathering:
+    - Global roles (system-wide permissions)
+    - Object roles (permissions on specific resources)
+    - Object metadata (organizations and teams the user has access to)
+
+    Args:
+        user: Django user model instance
+
+    Returns:
+        Dictionary containing:
+        - objects: Nested dict with arrays of organization/team objects user has access to
+        - object_roles: Dict mapping role names to content types and object indexes
+        - global_roles: List of global role names assigned to the user
+
+    Example:
+        {
+            'objects': {
+                'organization': [{'ansible_id': 'uuid1', 'name': 'Org1'}],
+                'team': [{'ansible_id': 'uuid2', 'name': 'Team1', 'org': 0}]
+            },
+            'object_roles': {
+                'Organization Admin': {'content_type': 'organization', 'objects': [0]}
+            },
+            'global_roles': ['Platform Auditor']
+        }
+    """
+    # Warm the DABContentType cache for efficient lookups
+    DABContentType.objects.warm_cache()
+
+    # Build object arrays and roles from user's assignments
+    object_arrays, object_roles = _build_objects_and_roles(user)
+
+    # Get global roles
+    global_roles = _get_user_global_roles(user)
+
+    # Build final claims structure
+    return {'objects': object_arrays, 'object_roles': object_roles, 'global_roles': global_roles}
+
+
+def get_user_claims_hashable_form(claims: dict) -> dict[str, Union[list[str], dict[str, list[str]]]]:
```

Comment:
is there a reason we can't just call this in get_claims_hash?

I expected the flow to be

```
claims = get_user_claims()
hashed = get_claims_hash(claims)
```

instead we have a call sandwiched in there

Reply:
What I think would be better-aligned with our eventual use is a method that gets the claims hash from a `user` object passed in. So that would be 3 internal calls, get claims, hashable form, then hash. I don't think we should combine pre-maturely, but adding another method that combines them would be totally valid.